lib/cw-core/src/tile.rs

```diff
@@ -88,9 +92,18 @@ impl Tile {
             return StatusCode::ErrorInternal;
         }
 
+        let different_frame_buffer = get_different_framebuffer(&mut self.previous_framebuffer,frame_buffer_unordered);
+
+        //check to use command 2 or command 3
+        if different_frame_buffer.len() / 5 < 100 {
+            let send_different_frame_buffer: &[u8] = &different_frame_buffer;
+            return self.command_3_update_specific_led(send_different_frame_buffer);
+        }
+
         // Generate framebuffer from pointer and generating the CRC by taking the sum of all the RGB values of the framebuffer
         let mut frame_buffer = [0; 1201];
         let mut crc: u8 = 0;
+
         for (i, byte) in frame_buffer_unordered.into_iter().enumerate() {
             crc += *byte;
             frame_buffer[self.index_converter_vector[i]] = *byte;
```

Comment:
The check if you are using command_3 or command_2 needs to happen **after** the ordering of the framebuffer. Otherwise all the indexes in the `send_different_frame_buffer` will be wrong.